examples/example.py

```diff
@@ -1,16 +1,121 @@
-''' Example how to interpret raw data and how to histogram the hits.
+'''Example how to use the M26 data interpreter. A hit table is created from raw data, additionally events are build using the TLU data words.
+    At the end, the hit table is formatted into the correct data format needed for testbeam analysis.
 '''
 
+import logging
+import os
+
+import numpy as np
 import tables as tb
 
+from tqdm import tqdm
+
+from contextlib2 import ExitStack
+
 from pyBAR_mimosa26_interpreter import data_interpreter
 
-# Example 1: How to use the build_hits function directly
-with tb.open_file('example_data_2.h5', 'r') as in_file_h5:
-    raw_data = in_file_h5.root.raw_data.read(8234, 8234 + 636 + 1000)
-    hits = data_interpreter.build_hits(raw_data)
 
-# Example 2: How to use the interpretation class to convert a raw data tabe
-with data_interpreter.DataInterpreter(raw_data_file='example_data_2.h5') as raw_data_analysis:
-    raw_data_analysis.interpret_word_table()
+logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(name)s - [%(levelname)-8s] (%(threadName)-10s) %(message)s")
+
+testbeam_analysis_dtype = np.dtype([
+    ('event_number', np.int64),
+    ('frame', np.uint32),
+    ('column', np.uint16),
+    ('row', np.uint16),
+    ('charge', np.uint16)])
+
+
+def process_dut(raw_data_file, output_filenames=None, trigger_data_format=2):
+    ''' Process and format raw data.
+
+    Parameters
+    ----------
+    raw_data_file : string
+        Filename of the raw data file.
+    output_filenames : list of strings
+        Filenames of the output interpreted and formatted data files.
+    trigger_data_format : int
+        Trigger/TLU FSM data mode.
+
+    Returns
+    -------
+    output_filenames : list of strings
+        Filenames of the output interpreted and formatted data files.
+    '''
+    analyzed_data_file = os.path.splitext(raw_data_file)[0] + '_interpreted.h5'
+    with data_interpreter.DataInterpreter(raw_data_file=raw_data_file, analyzed_data_file=analyzed_data_file, trigger_data_format=trigger_data_format, create_pdf=True) as mimosa_data_interpreter:
+        mimosa_data_interpreter.create_occupancy_hist = True
+        mimosa_data_interpreter.create_error_hist = True
+        mimosa_data_interpreter.create_hit_table = True
+        mimosa_data_interpreter.interpret_word_table()  # interpret raw data
+    output_filenames = format_hit_table(input_filename=analyzed_data_file, output_filenames=output_filenames)
+    return output_filenames
+
+
+def format_hit_table(input_filename, output_filenames=None, chunk_size=1000000):
```

Comment:
There is exactly the same code in testbeam_analysis/improvements branch right now:
https://github.com/SiLab-Bonn/testbeam_analysis/tree/improvements

We can clean that up if necessary. I kept a copy here, because we haven't modified all other branches of testbeam_analysis to be compatible with the new interpreter.